File: backend-quarkus/src/main/java/com/saas/entity/PdfJob.java
package com.saas.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "pdf_jobs")
public class PdfJob extends BaseEntity {

    @Column(name = "user_id", nullable = false)
    public UUID userId;

    @Column(nullable = false)
    public String status = "pending";

    @Column(name = "file_a_name", nullable = false)
    public String fileAName;

    @Column(name = "file_b_name", nullable = false)
    public String fileBName;

    @Column(name = "file_a_path", nullable = false)
    public String fileAPath;

    @Column(name = "file_b_path", nullable = false)
    public String fileBPath;

    @Column(name = "result_path")
    public String resultPath;

    @Column(name = "result_name")
    public String resultName;

    @Column(name = "error_message")
    public String errorMessage;

    public static List<PdfJob> findByTenant(UUID tenantId) {
        return find("tenantId = ?1 ORDER BY createdAt DESC", tenantId).list();
    }

    public static java.util.Optional<PdfJob> findByIdAndTenant(UUID id, UUID tenantId) {
        return find("id = ?1 AND tenantId = ?2", id, tenantId).firstResultOptional();
    }
}

File: backend-python/modules/pdf/schemas/pdf_job.py
from datetime import datetime

from pydantic import BaseModel

from enums.pdf_job_status import PdfJobStatus


class PdfJobOut(BaseModel):
    id: str
    tenant_id: str
    user_id: str
    status: PdfJobStatus
    file_a_name: str
    file_b_name: str
    result_name: str | None
    error_message: str | None
    created_at: datetime
    updated_at: datetime


def job_to_out(job: dict) -> PdfJobOut:
    return PdfJobOut(
        id=str(job["id"]),
        tenant_id=str(job["tenant_id"]),
        user_id=str(job["user_id"]),
        status=job["status"],
        file_a_name=job["file_a_name"],
        file_b_name=job["file_b_name"],
        result_name=job["result_name"],
        error_message=job["error_message"],
        created_at=job["created_at"],
        updated_at=job["updated_at"],
    )
